daemon/graphdriver/overlay2/overlay.go

```diff
@@ -312,17 +323,30 @@ func (d *Driver) Cleanup() error {
 // CreateReadWrite creates a layer that is writable for use as a container
 // file system.
 func (d *Driver) CreateReadWrite(id, parent string, opts *graphdriver.CreateOpts) error {
+	if opts != nil && len(opts.StorageOpt) != 0 && !projectQuotaSupported {
+		return fmt.Errorf("--storage-opt is supported only for overlay over xfs with 'pquota' mount option")
+	}
+
+	if opts == nil {
+		opts = &graphdriver.CreateOpts{
+			StorageOpt: map[string]string{},
+		}
+	}
+
+	if _, ok := opts.StorageOpt["size"]; !ok {
```

Comment:
Looks like we now _always_ pass `opts`, even if no option is set on `dockerd` or is passed with `docker run`; should the code here check if _either_ `opts.StorageOpt` is set _or_ `d.options.quota.Size` is set, and otherwise just keep `opts` as `nil`?